frame/support/src/traits/hooks.rs

```diff
@@ -165,27 +166,44 @@ pub trait OnRuntimeUpgrade {
 #[cfg_attr(all(feature = "tuples-96", not(feature = "tuples-128")), impl_for_tuples(96))]
 #[cfg_attr(feature = "tuples-128", impl_for_tuples(128))]
 impl OnRuntimeUpgrade for Tuple {
+	#[cfg(not(feature = "try-runtime"))]
 	fn on_runtime_upgrade() -> Weight {
 		let mut weight = Weight::zero();
 		for_tuples!( #( weight = weight.saturating_add(Tuple::on_runtime_upgrade()); )* );
 		weight
 	}
 
 	#[cfg(feature = "try-runtime")]
+	/// We are executing pre and post checks sequentially in order to be able to test seveal
+	/// consecutive migrations for the same pallet without errors. Therefore pre and post upgrade
+	/// hooks for tuples are a noop.
+	fn on_runtime_upgrade() -> Weight {
+		let mut weight = Weight::zero();
+		for_tuples!( #(
+			let _guard = frame_support::StorageNoopGuard::default();
+			// expected unwrap: we want to panic if any checks fail right here right now.
+			let state = Tuple::pre_upgrade().unwrap();
```

Comment:
Probably better to just have some new `on_runtime_upgrade_with_pre_and_post` that returns a result. (the name is for sure shit!)

Reply:
To be honest I'm not sure if needed, try-runtime is used for testing. If any of the checks panic - it's they need to be fixed. Seems like extra complexity to come up with another method that does have a Result return type just for the sake of a couple of unwraps. 

On the other hand it hurts readability a little bit. 

If I'm being honest I just don't want to add yet another method to this interface and make things more confusing for future implementors.

If I'm getting this right, you are talking about having an extra method on `OnRuntimeUpgrade` when `try-runtime`. 
The one that's going to be called instead of `on_runtime_upgrade`.

Alternatively we could introduce this method and still call it in `on_runtime_upgrade`, but then we'd have to unwrap anyway, so there is no point.

Reply:
If @bkchr's point is to only prevent `unwrap()`, I disagree. 

If there's more benefit in the new method, I am all ears.

Reply:
My point here was to prevent the unwrap. However, if we keep it. I demand a `expect` that provides more context, at least like the index of the tuple to have some idea what failed.

Reply:
We've discussed it before and I think the real answer here is `RUST_BACKTRACE=1`, which gives you exactly the failing line. That being said, @ruseinov can you please update the unwraps to expect?

Reply:
> My point here was to prevent the unwrap. However, if we keep it. I demand a `expect` that provides more context, at least like the index of the tuple to have some idea what failed.

Will do!